test/oai/test_client.py

```diff
@@ -151,10 +155,144 @@ def test_usage_summary():
+        shutil.rmtree(LEGACY_CACHE_DIR)
+
+    # Test default cache seed.
+    client = OpenAIWrapper(config_list=config_list)
+    start_time = time.time()
+    cold_cache_response = client.create(messages=[{"role": "user", "content": "random()"}])
+    end_time = time.time()
+    duration_with_cold_cache = end_time - start_time
+
+    start_time = time.time()
+    warm_cache_response = client.create(messages=[{"role": "user", "content": "random()"}])
+    end_time = time.time()
+    duration_with_warm_cache = end_time - start_time
+    assert cold_cache_response == warm_cache_response
+    assert duration_with_warm_cache < duration_with_cold_cache
+    assert os.path.exists(os.path.join(LEGACY_CACHE_DIR, str(LEGACY_DEFAULT_CACHE_SEED)))
+
+    # Test with cache seed set through constructor
+    client = OpenAIWrapper(config_list=config_list, cache_seed=13)
+    start_time = time.time()
+    cold_cache_response = client.create(messages=[{"role": "user", "content": "random()"}])
+    end_time = time.time()
+    duration_with_cold_cache = end_time - start_time
+
+    start_time = time.time()
+    warm_cache_response = client.create(messages=[{"role": "user", "content": "random()"}])
+    end_time = time.time()
+    duration_with_warm_cache = end_time - start_time
+    assert cold_cache_response == warm_cache_response
+    assert duration_with_warm_cache < duration_with_cold_cache
+    assert os.path.exists(os.path.join(LEGACY_CACHE_DIR, str(13)))
+
+    # Test with cache seed set through create method
+    client = OpenAIWrapper(config_list=config_list)
+    start_time = time.time()
+    cold_cache_response = client.create(messages=[{"role": "user", "content": "random()"}], cache_seed=17)
+    end_time = time.time()
+    duration_with_cold_cache = end_time - start_time
+
+    start_time = time.time()
+    warm_cache_response = client.create(messages=[{"role": "user", "content": "random()"}], cache_seed=17)
+    end_time = time.time()
+    duration_with_warm_cache = end_time - start_time
+    assert cold_cache_response == warm_cache_response
+    assert duration_with_warm_cache < duration_with_cold_cache
+    assert os.path.exists(os.path.join(LEGACY_CACHE_DIR, str(17)))
+
+    # Test using a different cache seed through create method.
+    start_time = time.time()
+    cold_cache_response = client.create(messages=[{"role": "user", "content": "random()"}], cache_seed=21)
+    end_time = time.time()
+    duration_with_cold_cache = end_time - start_time
+    assert duration_with_warm_cache < duration_with_cold_cache
+    assert os.path.exists(os.path.join(LEGACY_CACHE_DIR, str(21)))
+
+
+@pytest.mark.skipif(skip, reason="openai>=1 not installed")
+def test_cache():
+    config_list = config_list_from_json(
+        env_or_file=OAI_CONFIG_LIST,
+        file_location=KEY_LOC,
+        filter_dict={"model": ["gpt-3.5-turbo", "gpt-35-turbo"]},
+    )
+
+    # Clear cache.
+    if os.path.exists(LEGACY_CACHE_DIR):
```

Comment:
In future, provide utility function to clear cache. #1094

Reply:
Right.